qualtran/bloqs/factoring/mod_sub.py

```diff
@@ -138,7 +138,7 @@ def build_composite_bloq(self, bb: 'BloqBuilder', x: SoquetT) -> Dict[str, 'Soqu
         for i in range(self.bitsize):
             cvs = cvs + (0,)
         x_split = bb.split(x)
-        x_split, ctrl = bb.add(MultiControlX(cvs=cvs), ctrls=x_split, x=ctrl)
+        x_split, ctrl = bb.add(MultiControlPauli(cvs=cvs), controls=x_split, target=ctrl)
```

Comment:
the reader used to be able to see what was being done here and now there is ambiguity. Honestly, the most readable thing would be something like `XGate().controlled(CtrlSpec(cvs))`

Reply:
I agree, that would require overriding the `controlled()` method for `XGate()`, which can be done as a follow-up PR. 

The reason I am making this change is that I have plans to add more specialized and efficient decompositions for `MultiControlX`, which can also be used at the call sites instead of the more general `MultiControlPauli` that we use right now.

Reply:
isn't that a strong argument for keeping these separate

Reply:
Follow up decompositions for Multi control X would be implemented as a separate new Bloq; which isn't general enough for all pauli types

This Bloq works for all pauli types so it makes sense to have it this way.